meta/autoid/autoid.go

```diff
@@ -72,6 +72,39 @@ func (alloc *allocator) Alloc(tableID int64) (int64, error) {
 	return alloc.base, nil
 }
 
+var (
+	memID     int64
+	memIDLock sync.Mutex
+)
+
+type memoryAllocator struct {
+	mu    sync.Mutex
+	store kv.Storage
+	base  int64
+	end   int64
+	dbID  int64
+}
+
+// Alloc allocs the next autoID for table with tableID.
+// It gets a batch of autoIDs at a time. So it does not need to access storage for each call.
+func (alloc *memoryAllocator) Alloc(tableID int64) (int64, error) {
+	if tableID == 0 {
+		return 0, errors.New("Invalid tableID")
+	}
+	alloc.mu.Lock()
+	defer alloc.mu.Unlock()
+	if alloc.base == alloc.end { // step
+		memIDLock.Lock()
+		memID = memID + step
+		alloc.end = memID
+		alloc.base = alloc.end - step
```

Comment:
The above three lines of code can be understood as the following code ?
`alloc.end = memID + step
 alloc.base = memID`